six/modules/c++/six/source/Types.cpp

```diff
@@ -53,6 +53,11 @@ const char Constants::DES_USER_DEFINED_SUBHEADER_TAG[] = "XML_DATA_CONTENT";
 const char Constants::DES_USER_DEFINED_SUBHEADER_ID[] = "XML_DATA_CONTENT_773";
 const sys::Uint64_T Constants::DES_USER_DEFINED_SUBHEADER_LENGTH = 773;
 
+const char Constants::SICD_DESSHSI[] = 
```

Comment:
Update `NITFWriteControl` to use these too.  You may need to tweak how you did this a bit in order for the writer to use them...